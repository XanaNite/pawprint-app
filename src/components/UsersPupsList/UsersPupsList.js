import React from 'react'
import './UsersPupsList.css'
import UsersPup from '../UsersPup/UsersPup'

export default class UsersPupsList extends React.Component {
  render() {
    // const pups = this.props.pups.map((pup, i) => <UsersPup {...pup} key={i} />)
    this.props.pups.map(pup => console.log(pup))
    return (
      <>
        <ul>
          {this.props.pups.map((pup, i) => <UsersPup {...pup} key={i} />)}
        </ul>
      </>
    )
  }
}

UsersPupsList.defaultProps = {
  pups: [],
}