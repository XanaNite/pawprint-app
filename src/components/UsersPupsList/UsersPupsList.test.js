import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UsersPupsList from './UsersPupsList';

describe(`UsersPupsList component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders UsersPupsList by default', () => {
    const wrapper = shallow(<UsersPupsList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the UsersPupsList given props', () => {
    const wrapper = shallow(<UsersPupsList {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
