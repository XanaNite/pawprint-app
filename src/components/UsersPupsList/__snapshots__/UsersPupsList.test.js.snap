// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPupsList component renders UsersPupsList by default 1`] = `
<Fragment>
  <ul />
</Fragment>
`;

exports[`UsersPupsList component renders the UsersPupsList given props 1`] = `
<Fragment>
  <ul />
</Fragment>
`;
